File: library/src/main/java/cn/cqray/android/ui/multi/MultiFragmentAdapter.java
package cn.cqray.android.ui.multi;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import java.util.List;

/**
 * Fragment适配器
 * @author Cqray
 */
public class MultiFragmentAdapter extends FragmentStateAdapter {

    private final List<Fragment> fragmentList;

    public MultiFragmentAdapter(@NonNull Fragment fragment, List<Fragment> fragmentList) {
        super(fragment);
        this.fragmentList = fragmentList;
    }

    public MultiFragmentAdapter(@NonNull FragmentActivity activity, List<Fragment> fragmentList) {
        super(activity);
        this.fragmentList = fragmentList;
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        return fragmentList.get(position);
    }

    @Override
    public int getItemCount() {
        return fragmentList.size();
    }

    @NonNull
    public List<Fragment> getFragmentList() {
        return fragmentList;
    }
}

File: library/src/main/java/cn/cqray/android/ui/multi/MultiFragmentAdapter.kt
package cn.cqray.android.ui.multi

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter

/**
 * Fragment适配器
 * @author Cqray
 */
class MultiFragmentAdapter : FragmentStateAdapter {

    val fragments: MutableList<Fragment> = ArrayList()

    constructor(fragment: Fragment, fragments: List<Fragment>) : super(fragment) {
        this.fragments.addAll(fragments)
    }

    constructor(activity: FragmentActivity, fragments: List<Fragment>) : super(activity) {
        this.fragments.addAll(fragments)
    }

    override fun createFragment(position: Int): Fragment {
        return fragments[position]
    }

    override fun getItemCount(): Int {
        return fragments.size
    }
}